drivers/sdhc/rcar_mmc.c

```diff
@@ -1613,6 +1647,33 @@ static int rcar_mmc_init_controller_regs(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_RCAR_MMC_DMA_IRQ_DRIVEN_SUPPORT
+/**
+ * @brief MMC IRQ handler
+ *
+ * @param arg The pointer to an MMC device structure
+ *
+ * @retval none
+ */
+static void rcar_mmc_irq_handler(const void *arg)
+{
+	const struct device *dev = arg;
+
+	uint32_t dma_info1 = rcar_mmc_read_reg32(dev, RCAR_MMC_DMA_INFO1);
+	uint32_t dma_info2 = rcar_mmc_read_reg32(dev, RCAR_MMC_DMA_INFO2);
+
+	if (dma_info1 || dma_info2) {
+		struct mmc_rcar_data *data = dev->data;
+
+		rcar_mmc_write_reg32(dev, RCAR_MMC_DMA_INFO1_MASK, 0xfffffeff);
+		rcar_mmc_write_reg32(dev, RCAR_MMC_DMA_INFO2_MASK, ~0);
+		k_sem_give(&data->irq_xref_fin);
+	} else {
+		LOG_WRN("%s: warning: non-dma event triggers irq", dev->name);
```

Comment:
Be careful with printing in IRQ context, previously we've faced with some issues related to this. If it's not critical, I propose to omit this.

Reply:
Which kind of problems have you seen?

I have been debugging the IRQ handler using prints and haven't caught any issues during debug

Looks like `LOG_` API may work from IRQ context https://docs.zephyrproject.org/latest/services/logging/index.html:
```
Default frontend is engaged when logging API is called in a source of logging (e.g. LOG_INF) and is responsible for filtering a message (compile and run time), allocating buffer for the message, creating the message and committing that message. Since logging API can be called in an interrupt, frontend is optimized to log the message as fast as possible.
```